backend/spec/integration/membersIntegrationSpec.js

```diff
@@ -131,19 +132,24 @@ describe('MemberIntegrationTests', () => {
   });
 
   describe('Edit member', () => {
-    beforeEach(() => integrationTestHelpers.resetDatabase()
-            .then(integrationTestHelpers.createBranch)
-            .then(setState(browserState, 'branch'))
-            .then(branch => { integrationTestHelpers.createBranchAdmin(branch); return branch; })
-            .then(integrationTestHelpers.createMembers(agent, 1))
-            .then(() => Q.all([
-              integrationTestHelpers.createGroup(agent, browserState.branch.id),
-              integrationTestHelpers.createGroup(agent, browserState.branch.id),
-            ]))
-            .then(setState(browserState, 'groups'))
-            .then(integrationTestHelpers.authenticateBranchAdmin(agent))
-            .then(getMembers(agent, browserState))
-            .then(setState(browserState, 'members')));
+    beforeEach(() => (
+      integrationTestHelpers.resetDatabase()
+        .then(integrationTestHelpers.createBranch)
+        .then(setState(browserState, 'branch'))
+        .then(branch => (
+          integrationTestHelpers.createBranchAdmin(branch)
+            .then(() => branch)
```

Comment:
This extra nested `then` is where the actual fix is. Previously the promise chain would sometimes move on before the branch admin had been created.